### PR TITLE
Splitting transitions into backdrop and modal

### PR DESCRIPTION
**Pull request by mlunoe**

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @jfurrow

File: src/Modal/ModalContents.js [hunk 1]
```diff
@@ -222,34 +223,52 @@ export default class ModalContents extends Util.mixin(BindMixin) {
...
+    return (
+      <div className={props.backdropClass} onClick={this.handleBackdropClick} />
+    );
+  }
+
   render() {
+    let props = this.props;
+    let containerClass = {};
+    if (props.open) {
+      containerClass[props.containerClass] = true;
```
Comment (rcorral) [inline src/Modal/ModalContents.js]: Why this change?

File: src/Modal/ModalContents.js [hunk 2]
```diff
@@ -222,34 +223,52 @@ export default class ModalContents extends Util.mixin(BindMixin) {
...
+  getBackdrop() {
+    let props = this.props;
+    if (!props.open) {
+      return null;
+    }
+
+    return (
+      <div className={props.backdropClass} onClick={this.handleBackdropClick} />
+    );
+  }
```
Comment (rcorral) [inline src/Modal/ModalContents.js]: Can a test be written for this?

File: src/Modal/ModalContents.js (hunk 1 above)
Comment (rcorral) [inline src/Modal/ModalContents.js]: And can we test it.

Comment (mlunoe) [inline src/Modal/ModalContents.js]: It has to change, because we can't apply the classes when it is not open. Otherwise it will cover over the whole page all the time

Comment (mlunoe) [inline src/Modal/ModalContents.js]: Tests added

File: src/Modal/ModalContents.js (hunk 2 above)
Comment (mlunoe) [inline src/Modal/ModalContents.js]: done

**Comment by jfurrow** [comment]
This looks good! Will this affect the way we animate modals in DCOS UI?

**Comment by mlunoe** [comment]
Nope, it will stay the same!

**Comment by jfurrow** [comment]
Cool, this is great!

**Comment by jfurrow** [comment]
:ship:
